src/agents_ets.erl

```diff
@@ -107,3 +82,19 @@ clean(TableId) ->
     catch
         _:_ -> error
     end.
+
+-spec get_list(agents()) ->
+                      [agent()].
+get_list(Agents) ->
+    ets:tab2list(Agents).
+
+
+-spec get_tiles(pos_integer(), environment()) -> [{dimension(), tile()}].
+get_tiles(Dist, Env) ->
+    W = (Env#env.world)#world.w,
+    TagTiles = [{I, Env#env.agents} || I <- lists:seq(1, W, Dist)],
+    TagTiles.
+
+-spec update_tiles({tile(), environment()}, environment(), config()) -> environment().
```

Comment:
Line 98 is too long: -spec update_tiles({tile(), environment()}, environment(), config()) -> environment()..